File: ng-Gomax/src/app/Data-Processing/show/models/show.ts
import {RoomShow} from '../../room/models/roomShow';
import {Film} from '../../film/models/film';
import {Cinema} from '../../cinema/models/cinema';

export class Show {

  constructor(
    /*tslint:disable:variable-name*/
    public id: number,
    public startTime?: string,
    public endTime?: string,
    public status?: number,
    public film?: Film,
    public cinema?: Cinema,
    public roomShow?: RoomShow,
  ) {
  }
}

File: sp-gomax/src/main/java/com/gomax/gmx/entities/Show.java
package com.gomax.gmx.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "_show")
public class Show {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Timestamp startTime;

    private Timestamp endTime;

    private long status = 0;

    @ManyToOne
    private Film film;

    @OneToOne
    private Cinema cinema;

    @OneToOne
    private RoomShow roomShow;

}
